Habitica Intents/TaskManager.swift

```diff
@@ -0,0 +1,179 @@
+//
+//  TaskManager.swift
+//  Habitica
+//
+//  Created by Christopher Coffin on 3/12/19.
+//  Copyright © 2019 HabitRPG Inc. All rights reserved.
+//
+
+import Foundation
+import Habitica_API_Client
+import Habitica_Database
+import ReactiveSwift
+import Habitica_Models
+//import Result
+
+
+class TaskManager: BaseRepository<TaskLocalRepository>, TaskRepositoryProtocol {
+    static let shared = TaskManager()
+    
+    override init() {
+        super.init()
+        self.setupNetworkClient()
+        if let cid = AuthenticationManager.shared.currentUserId {
+            AuthenticationManager.shared.currentUserKey = AuthenticationManager.shared.localKeychain[cid]
+        }
+        print("Auth at start id, ", AuthenticationManager.shared.currentUserId)
+        print("Auth at start key, ", AuthenticationManager.shared.currentUserKey)
+    }
+    
+    @objc
+    func setupNetworkClient() {
+        /*
```

Comment:
The issue here is that the AppDelegate is not called when the intent is launched, right? So that you copied that code here, so that it could run correctly?

Reply:
Yes, it's entirely possible that there's a better solution, but not one that I'm aware of.